GCRCatalogs/dc2_object.py

```diff
@@ -212,7 +213,13 @@ def _subclass_init(self, **kwargs):
         self.base_dir = kwargs['base_dir']
         self._filename_re = re.compile(kwargs.get('filename_pattern', FILE_PATTERN))
         self._groupname_re = re.compile(kwargs.get('groupname_pattern', GROUP_PATTERN))
-        self._schema_path = kwargs.get('schema_path', os.path.join(self.base_dir, SCHEMA_PATH))
+
+        _schema_filename = kwargs.get('schema_filename', SCHEMA_FILENAME)
+        if os.path.isabs(_schema_filename):
+            self._schema_path = _schema_filename
+        else:
+            self._schema_path = os.path.join(self.base_dir, _schema_filename)
```

Comment:
There is in fact no need to check if `_schema_filename` is absolute path, because the behavior of `os.path.join` is that everything before an absolute path will be ignored ([ref](https://docs.python.org/3.6/library/os.path.html#os.path.join)). 

Hence, you can safely replace Lines 218-221 with just
```python
self._schema_path = os.path.join(self.base_dir, _schema_filename)
```

Reply:
Woohoo!